CRM/Utils/Mail/EmailProcessor.php

```diff
@@ -94,14 +93,10 @@ public static function process($civiMail = TRUE) {
   public static function _process($civiMail, $dao, $is_create_activities) {
     // 0 = activities; 1 = bounce;
     $usedfor = $dao->is_default;
-
-    $emailActivityTypeId
-      = (defined('EMAIL_ACTIVITY_TYPE_ID') && EMAIL_ACTIVITY_TYPE_ID)
-      ? EMAIL_ACTIVITY_TYPE_ID
-      : CRM_Core_PseudoConstant::getKey('CRM_Activity_BAO_Activity', 'activity_type_id', 'Inbound Email');
-
-    if (!$emailActivityTypeId) {
-      throw new CRM_Core_Exception(ts('Could not find a valid Activity Type ID for Inbound Email'));
+    if ($usedfor == 0) {
+      // create an array of all of to, from, cc, bcc that are in use for this Mail Account, so we don't create contacts for emails we aren't adding to the activity
+      $emailFields = array_unique(array_merge(explode(",", $dao->activity_targets), explode(",", $dao->activity_assignees), explode(",", $dao->activity_source)));
```

Comment:
If I tell it I don't want any assignees, then $dao->activity_assignees is the empty string, which exploded gives array(0 => ''), which then later tries to look for `$mail->''` which gives a fatal error.

But also array_merge doesn't seem right here, because it's merging based on (semi-random) numeric keys, which doesn't actually "merge" because (a) it's not the values, and (b) php is either weird or awesome depending on your point of view and it won't merge numeric keys.

Reply:
Yes, intentionally doing the weird PHP thing here. Open to better ways though.

Maybe it would be best to use a SEPARATOR_BOOKEND or some other serialization for these fields?

Reply:
I suppose you'd still have to explode since you'll always have a string to start with. Could add an array_filter()?

Reply:
It doesn't look pretty, but it works with empty values now. Thanks for catching that.